apps/rush-lib/src/utilities/RushCiMode.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as isCI from 'is-ci';
+import * as colors from 'colors';
+
+/**
+ * Determines if Rush is running in CI mode or NON-CI mode
+ */
+export class RushCiMode {
+
+  private static _rushCiMode: RushCiMode;
+
+  private readonly _forceCI: boolean;
+  private readonly _forceNONCI: boolean;
+
+  private static _initialized: boolean;
+
+  private constructor(overrideMode: string | undefined) {
+    this._forceCI = false;
+    this._forceNONCI = false;
+    if (!overrideMode) {
+      // Nothing to do.
+      return;
+    }
+
+    if (overrideMode === 'CI') {
+      console.log(colors.yellow('WARNING: Rush mode overriden to: ' + overrideMode));
+      this._forceCI = true;
+    } else if (overrideMode === 'NON-CI') {
+      console.log(colors.yellow('WARNING: Rush mode overriden to: ' + overrideMode));
+      this._forceNONCI = true;
+    } else {
+      console.log(colors.yellow('WARNING: Ignoring unknown Rush mode override string: ' + overrideMode));
```

Comment:
This should be a hard error.

Reply:
Will that make working with older versions difficult?

Reply:
That env variable shouldn't be set in older versions. It would have caused an error already since we error when an unknown `RUSH_...` env variable is set.

Reply:
If someone has multiple rush repos each configured to use a different version of Rush and one of them supports an environment variable (for example RUSH_PNPM_STORE_PATH) and the other two don't, then wouldn't getting hard errors for this on the other two be painful?